qiita_pet/handlers/qiita_redbiom.py

```diff
@@ -0,0 +1,110 @@
+            contexts = df.ContextName.values
+            if search_on == 'metadata':
+                query = query.lower()
+                try:
+                    samples = redbiom.search.metadata_full(query, False)
+                except TypeError:
+                    message = (
+                        'Not a valid search: "%s", are you sure this is a '
+                        'valid metadata %s?' % (
+                            query, 'value' if search_on == 'metadata' else
+                            'category'))
+                except ValueError:
+                    message = (
+                        'Not a valid search: "%s", your query is too small '
+                        '(too few letters), try a longer query' % query)
+                if message == '':
+                    sids = set([s.split('.', 1)[0] for s in samples])
+                    for s in sids:
+                        study_artifacts[s] = [a.id for a in Study(s).artifacts(
+                            artifact_type='BIOM')]
+            elif search_on == 'feature':
+                query = [f for f in query.split(' ')]
+                for ctx in contexts:
+                    for idx in redbiom.util.ids_from(query, True, 'feature',
+                                                     ctx):
+                        aid, sid = idx.split('_', 1)
+                        sid = sid.split('.', 1)[0]
+                        study_artifacts[sid].append(aid)
+            elif search_on == 'taxon':
+                for ctx in contexts:
+                    # find the features with those taxonomies and then search
+                    # those features in the samples
+                    features = redbiom.fetch.taxon_descendents(ctx, query)
+                    for idx in redbiom.util.ids_from(features, True, 'feature',
+                                                     ctx):
+                        aid, sid = idx.split('_', 1)
+                        sid = sid.split('.', 1)[0]
+                        study_artifacts[sid].append(aid)
+            else:
+                message = ('Incorrect search by: you can use metadata, '
+                           'features or taxon and you passed: %s' % search_on)
+
+            if message == '':
+                keys = study_artifacts.keys()
+                if keys:
+                    results = generate_study_list_without_artifacts(
+                        study_artifacts.keys(), True)
+                    # inserting the artifact_biom_ids to the results
+                    for i in range(len(results)):
+                        results[i]['artifact_biom_ids'] = list(set(
+                            study_artifacts[str(results[i]['study_id'])]))
+                else:
+                    message = "No samples where found! Try again ..."
+
+        callback((results, message))
+
+    @coroutine
+    @execute_as_transaction
+    def post(self, search):
+        search = self.get_argument('search', None)
```

Comment:
It looks like `search` and `search_on` are not really optional parameters.
I think the correct thing to do here is:
```python
search = self.get_argument('search')
search_on = self.get_argument('search_on')
```
and let tornado handle if the arguments are not passed. It will return a 400 error message which is the correct code given that the request is incorrect.